hail/src/main/scala/is/hail/sparkextras/MultiWayJoinRDD.scala

```diff
@@ -0,0 +1,49 @@
+package is.hail.sparkextras
+
+import org.apache.spark.{OneToOneDependency, Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+object MultiWayJoinRDD {
+  def apply[T: ClassTag , V: ClassTag](
+    rdds: IndexedSeq[RDD[T]],
+    preservesPartitioning: Boolean = false
+  )(f: (Array[Iterator[T]]) => Iterator[V]): MultiWayJoinRDD[T, V] = {
+    new MultiWayJoinRDD(rdds.head.sparkContext, rdds, f, preservesPartitioning)
+  }
+}
+
+private case class MultiWayJoinPartition(val index: Int, val partitions: IndexedSeq[Partition])
+  extends Partition
+
+class MultiWayJoinRDD[T: ClassTag, V: ClassTag](
```

Comment:
Minor point, but to be consistent with the rest of our codebase, I would name this `MultiWayZipPartitionsRDD`.

Reply:
I've gone with `MultiWayZipRDD` in the other PR for now. I guess to be consistent with spark, I should use `MultiWayZipPartitionsRDD` analogue to `ZippedPartitionsRDD`. I'll wait until I'm staging this on top of the first PR, which is now approved though.